### PR TITLE
Search: Remove dependency on @wordpress/interface

### LINKED ISSUE #21065: Search: Search Editor Broken after Monorepo Update
### Impacted plugin

Jetpack

### Steps to Reproduce

- Spin up a site with the Jetpack beta and Jetpack Search subscription.
- Set the beta to Bleeding Edge
- Visit the Jetpack Search customization page.	

### A clear and concise description of what you expected to happen.

The customization page to load.

### What actually happened

The screen didn't load - just a white screen with a console error:

```
Uncaught TypeError: y is undefined
    J https://continental-tick.jurassic.ninja/wp-content/plugins/jetpack-dev/_inc/build/instant-search/jp-search-configure-main.min.js?ver=10.2-alpha:14
    we https://continental-tick.jurassic.ninja/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.13.1:84
```

### Other information

Seems it may have been a result of updating the Monorepo from this PR: https://github.com/Automattic/jetpack/pull/21021

Additional information: p1631565793050800-slack-CBG1CP4EN

### Operating System

_No response_

### OS Version

_No response_

### Browser

Firefox

### Browser Version(s)

_No response_

### PR DESCRIPTION
**Pull request by jsnmoon**
Fixes #21065.

#### Changes proposed in this Pull Request:
* Removes `@wordpress/interface`.
* Removes usage of the interface package in Customberg.
* The Customberg sidebar is no longer dismissable in desktop breakpoints.
* The Customberg sidebar is now open by default on mobile breakpoints.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Apply this change to your Jetpack site with a Search subscription.
* Open Customberg (`/wp-admin/admin.php?page=jetpack-search-configure`).
* Ensure that the sidebar is always open on desktop viewports.
* Resize the window to mimic a mobile breakpoint and refresh the page.
* Ensure that the page opens with the sidebar open by default.

### CODE REVIEW

File: projects/plugins/jetpack/modules/search/customberg/components/header/index.jsx [hunk 1]
```diff
@@ -36,7 +35,14 @@ function Header() {
 			</div>
 			<div className="jp-search-configure-header__actions">
 				<SaveButton />
-				<PinnedItems.Slot scope={ COMPLEMENTARY_AREA_SCOPE } />
+				<Button
+					ariaLabel={ __( 'Show settings', 'jetpack' ) }
```
Comment (kangzj) [inline projects/plugins/jetpack/modules/search/customberg/components/header/index.jsx:39]: should `ariaLabel` here be `aria-label`?

File: projects/plugins/jetpack/modules/search/customberg/components/sidebar/index.jsx [hunk 2]
```diff
@@ -1,113 +1,47 @@
...
-			{ currentTab === OPTIONS_TAB_IDENTIFIER && <SidebarOptions /> }
-		</ComplementaryArea>
+		<div className="interface-complementary-area jp-search-configure-sidebar">
+			<div
+				className="components-panel__header interface-complementary-area-header jp-search-configure-sidebar__panel-tabs"
+				tabindex="-1"
+			>
+				<Tabs enabledSidebarName={ enabledSidebarName } enableSidebar={ enableSidebar } />
+				<Button
+					ariaLabel={ __( 'Close settings', 'jetpack' ) }
```
Comment (kangzj) [inline projects/plugins/jetpack/modules/search/customberg/components/sidebar/index.jsx:33]: should `ariaLabel` here be `aria-label`?

**Review by kangzj — DISMISSED** [review]
Good job @jsnmoon Thanks for the fast fix 👍 
I tested locally and also on JN. It works well and looks good to me.
1. Two comments regarding `aria-label`
2. Probably make the linter happy before shippping?

but they are not a blockers.

File: projects/plugins/jetpack/modules/search/customberg/components/sidebar/index.jsx (hunk 2 above)
Comment (bluefuton) [reply projects/plugins/jetpack/modules/search/customberg/components/sidebar/index.jsx:33]: Yep - oddly these aren't camel-cased in React:

https://reactjs.org/docs/accessibility.html

**Comment by anomiex** [comment]
>     2. Probably make the linter happy before shippping?
> 
> but they are not a blockers.

Making the linter happy is a blocker, in that the "Linting / ESLint (non-excluded files only)" job is marked as required so GitHub won't allow a merge until it is passing.

**Comment by kangzj** [comment]
> > ```
> > 2. Probably make the linter happy before shippping?
> > ```
> >     
> >   
> > but they are not a blockers.
> 
> Making the linter happy is a blocker, in that the "Linting / ESLint (non-excluded files only)" job is marked as required so GitHub won't allow a merge until it is passing.

Hi @anomiex, thanks for pointing that out. TIL. 👍

**Comment by jsnmoon** [comment]
Lint errors have been fixed.

Aside: I noticed that my ESLint integration with VSCode has stopped working some time ago. Not sure why, but I needed to run `pnpm lint-required` to manually suss out what I needed to fix.

File: projects/plugins/jetpack/modules/search/customberg/components/header/index.jsx (hunk 1 above)
Comment (jsnmoon) [reply projects/plugins/jetpack/modules/search/customberg/components/header/index.jsx:41]: Nice catch!

File: projects/plugins/jetpack/modules/search/customberg/components/sidebar/index.jsx (hunk 2 above)
Comment (jsnmoon) [reply projects/plugins/jetpack/modules/search/customberg/components/sidebar/index.jsx]: 👍

**Comment by bluefuton** [comment]
This is a big improvement to the mobile user experience and would say it also fixes https://github.com/Automattic/jetpack/issues/20908. Nicely done :)

A couple of minor things I noticed - we now get a double scroll bar when the preview is collapsed:

Screen Shot 2021-09-16 at 12 09 12

A strange minor bug popped up in Edge, where the breadcrumb is scrollable (this only happens in Customberg not the front end of the site):

Screen Shot 2021-09-16 at 12 10 32

**Review by anomiex — APPROVED** [review]
From a code perspective it looks ok to me, although I do hope you'll iterate on `projects/plugins/jetpack/modules/search/customberg/lib/wordpress-interface.scss` as the TODO comment suggests.

I don't know if the intention is to iterate on the scrollbar issues pointed out or to fix them before merging this. IMO it'd be good to merge this to have the feature mostly functional (so it no longer blocks the weekly Atomic deploys we're trying to get going, for example) and iterate on the minor issues.

**Comment by jsnmoon** [comment]
🚢 r231933-wpcom

**Pull request merged**